lib/components/InputBox/BaseInputBox.tsx

```diff
@@ -40,17 +42,49 @@ const BaseInputBox: FC<BaseInputBoxProps> = ({
   ...inputProps
 }) => {
   const [inputID] = useState(inputProps?.id || uniqueid("pebbles__input__"));
+  const [isPassword, setIsPassword] = useState(false);
+  const [showPwd, setShowPwd] = useState(false);
+  const [inputType, setInputType] = useState<HTMLInputTypeAttribute>();
+
+  useEffect(() => {
+    inputProps?.type === "password" && setIsPassword(true);
+    inputProps.type && setInputType(inputProps.type);
+  }, []);
+
+  useEffect(() => {
+    if (isPassword) {
+      showPwd ? setInputType("text") : setInputType("password");
+    }
+  }, [showPwd, isPassword]);
+
+  const switchShowPwd = () => {
+    setShowPwd((prevState) => !prevState);
+  };
 
   return (
     <StyledDiv spaced={spaced}>
       <Label htmlFor={inputID}>{description}</Label>
       <InputDiv>
         {!inputProps?.children && (
-          <StyledInput status={status} {...inputProps} id={inputID} />
+          <StyledInput
+            status={status}
+            {...inputProps}
+            type={inputType}
```

Comment:
you can just do:
type = { showPassword ? "text" : inputProps.type }
keep it simple

Reply:
I will check all the proposed reviews, see if I can make it work like that

Reply:
Update: yes, this line of code made things much simples, without the useState/useEffect stuff. Thanks Felipe ;-)